src/lexer.rs

```diff
@@ -633,20 +625,24 @@ pub fn lex(input: &str, remove_trailing_operator: bool, default_degree: Unit) ->
   }
   let tokens = &mut lexer.tokens;
   // auto insert missing parentheses in first and last position
-  if lexer.left_paren_count > lexer.right_paren_count {
-    let missing_right_parens = lexer.left_paren_count - lexer.right_paren_count;
-    for _ in 0..missing_right_parens {
-      tokens.push(Token::Operator(RightParen));
+  match lexer.left_paren_count.cmp(&lexer.right_paren_count) {
+    Ordering::Less => {
+      let missing_left_parens = lexer.right_paren_count - lexer.left_paren_count;
+      for _ in 0..missing_left_parens {
+        tokens.insert(0, Token::Operator(LeftParen));
+      }
     }
-  } else if lexer.left_paren_count < lexer.right_paren_count {
-    let missing_left_parens = lexer.right_paren_count - lexer.left_paren_count;
-    for _ in 0..missing_left_parens {
-      tokens.insert(0, Token::Operator(LeftParen));
+    Ordering::Equal => {}
+    Ordering::Greater => {
+      let missing_right_parens = lexer.left_paren_count - lexer.right_paren_count;
+      for _ in 0..missing_right_parens {
+        tokens.push(Token::Operator(RightParen));
+      }
```

Comment:
To me using `.cmp()` seems harder to read than comparison operators, but maybe it's fine?

Reply:
It's definitely not as straightforward to understand at first glance.